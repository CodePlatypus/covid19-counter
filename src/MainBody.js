import TotalStatBody from './TotalStatBody';
import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import CardBody from './CardBody';
import TableBody from './TableBody';
import NewsBox from './NewsBox';
import PropTypes from 'prop-types';

export default function MainBody(props) {
    const {
        isLoading,
        totalStat,
        countriesStat,
        apNews,
        windowWidth
    } = props;
    return (
        <div>
            <hr/>
            <TotalStatBody isLoading={isLoading} totalStat={totalStat}/>
            <hr/>
            {isLoading ? <CircularProgress/>
                : (isMobile(windowWidth) ? <CardBody countriesStat={countriesStat}/> :
                    <TableBody countriesStat={countriesStat}/>)
            }
            <hr/>
            {isLoading &&
            apNews.map(newsItem => <NewsBox key={newsItem.title} isLoading={isLoading}
                                                       newsItem={newsItem}></NewsBox>)
            }
        </div>
    )
}

const isMobile = (windowWidth) => {
    return windowWidth <= 750;
};


MainBody.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    totalStat: PropTypes.array.isRequired,
    countriesStat: PropTypes.array.isRequired,
    apNews: PropTypes.array.isRequired,
    windowWidth: PropTypes.number.isRequired
};