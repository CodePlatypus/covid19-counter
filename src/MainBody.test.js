import React from 'react';
import {shallow} from 'enzyme';
import MainBody from './MainBody';
import toJson from 'enzyme-to-json';

describe('Snapshots for CardBody with CircularProgress when isLoading', () => {
    test('renders CardBody with only CircularProgress when isLoading desktop', () => {
        const wrapper = shallow(<MainBody isLoading={true}
                                          countriesStat={countriesStat}
                                          totalStat={totalStat}
                                          apNews={apNews}
                                          windowWidth={751}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('Snapshots for CardBody not isLoading', () => {

    test('renders CardBody when not isLoading desktop', () => {
        const wrapper = shallow(<MainBody isLoading={false}
                                          countriesStat={countriesStat}
                                          totalStat={totalStat}
                                          apNews={apNews}
                                          windowWidth={751}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders CardBody when not isLoading mobile1', () => {
        const wrapper = shallow(<MainBody isLoading={false}
                                          countriesStat={countriesStat}
                                          totalStat={totalStat}
                                          apNews={apNews}
                                          windowWidth={750}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders CardBody  when not isLoading mobile2', () => {
        const wrapper = shallow(<MainBody isLoading={false}
                                          countriesStat={countriesStat}
                                          totalStat={totalStat}
                                          apNews={apNews}
                                          windowWidth={749}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders CardBody no news when not isLoading desktop', () => {
        const wrapper = shallow(<MainBody isLoading={false}
                                          countriesStat={countriesStat}
                                          totalStat={totalStat}
                                          apNews={[]}
                                          windowWidth={751}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

const totalStat = {
    "cases": 254701,
    "deaths": 10447,
    "recovered": 89071,
    "updated": 1584710988792
};

const countriesStat = [{
    "country": "China",
    "cases": 80967,
    "todayCases": 39,
    "deaths": 3248,
    "todayDeaths": 3,
    "recovered": 71150,
    "activeCases": 6569,
    "critical": 2136
}, {
    "country": "Italy",
    "cases": 12313,
    "todayCases": 234,
    "deaths": 123,
    "todayDeaths": 23,
    "recovered": 32523,
    "activeCases": 4324,
    "critical": 2345
}];

const apNews = [{
    "title": "Title1",
    "description": "description1",
    "link": "https://link1.com"
}, {
    "title": "Title2",
    "description": "description2",
    "link": "https://link2.com"
}];