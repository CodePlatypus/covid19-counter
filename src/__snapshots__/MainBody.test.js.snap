// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots for CardBody not isLoading renders CardBody  when not isLoading mobile2 1`] = `
<div>
  <hr />
  <TotalStatBody
    isLoading={false}
    totalStat={
      Object {
        "cases": 254701,
        "deaths": 10447,
        "recovered": 89071,
        "updated": 1584710988792,
      }
    }
  />
  <hr />
  <CardBody
    countriesStat={
      Array [
        Object {
          "activeCases": 6569,
          "cases": 80967,
          "country": "China",
          "critical": 2136,
          "deaths": 3248,
          "recovered": 71150,
          "todayCases": 39,
          "todayDeaths": 3,
        },
        Object {
          "activeCases": 4324,
          "cases": 12313,
          "country": "Italy",
          "critical": 2345,
          "deaths": 123,
          "recovered": 32523,
          "todayCases": 234,
          "todayDeaths": 23,
        },
      ]
    }
  />
  <hr />
  <NewsBox
    isLoading={false}
    key="Title1"
    newsItem={
      Object {
        "description": "description1",
        "link": "https://link1.com",
        "title": "Title1",
      }
    }
  />
  <NewsBox
    isLoading={false}
    key="Title2"
    newsItem={
      Object {
        "description": "description2",
        "link": "https://link2.com",
        "title": "Title2",
      }
    }
  />
</div>
`;

exports[`Snapshots for CardBody not isLoading renders CardBody no news when not isLoading desktop 1`] = `
<div>
  <hr />
  <TotalStatBody
    isLoading={false}
    totalStat={
      Object {
        "cases": 254701,
        "deaths": 10447,
        "recovered": 89071,
        "updated": 1584710988792,
      }
    }
  />
  <hr />
  <TableBody
    countriesStat={
      Array [
        Object {
          "activeCases": 6569,
          "cases": 80967,
          "country": "China",
          "critical": 2136,
          "deaths": 3248,
          "recovered": 71150,
          "todayCases": 39,
          "todayDeaths": 3,
        },
        Object {
          "activeCases": 4324,
          "cases": 12313,
          "country": "Italy",
          "critical": 2345,
          "deaths": 123,
          "recovered": 32523,
          "todayCases": 234,
          "todayDeaths": 23,
        },
      ]
    }
  />
  <hr />
</div>
`;

exports[`Snapshots for CardBody not isLoading renders CardBody when not isLoading desktop 1`] = `
<div>
  <hr />
  <TotalStatBody
    isLoading={false}
    totalStat={
      Object {
        "cases": 254701,
        "deaths": 10447,
        "recovered": 89071,
        "updated": 1584710988792,
      }
    }
  />
  <hr />
  <TableBody
    countriesStat={
      Array [
        Object {
          "activeCases": 6569,
          "cases": 80967,
          "country": "China",
          "critical": 2136,
          "deaths": 3248,
          "recovered": 71150,
          "todayCases": 39,
          "todayDeaths": 3,
        },
        Object {
          "activeCases": 4324,
          "cases": 12313,
          "country": "Italy",
          "critical": 2345,
          "deaths": 123,
          "recovered": 32523,
          "todayCases": 234,
          "todayDeaths": 23,
        },
      ]
    }
  />
  <hr />
  <NewsBox
    isLoading={false}
    key="Title1"
    newsItem={
      Object {
        "description": "description1",
        "link": "https://link1.com",
        "title": "Title1",
      }
    }
  />
  <NewsBox
    isLoading={false}
    key="Title2"
    newsItem={
      Object {
        "description": "description2",
        "link": "https://link2.com",
        "title": "Title2",
      }
    }
  />
</div>
`;

exports[`Snapshots for CardBody not isLoading renders CardBody when not isLoading mobile1 1`] = `
<div>
  <hr />
  <TotalStatBody
    isLoading={false}
    totalStat={
      Object {
        "cases": 254701,
        "deaths": 10447,
        "recovered": 89071,
        "updated": 1584710988792,
      }
    }
  />
  <hr />
  <CardBody
    countriesStat={
      Array [
        Object {
          "activeCases": 6569,
          "cases": 80967,
          "country": "China",
          "critical": 2136,
          "deaths": 3248,
          "recovered": 71150,
          "todayCases": 39,
          "todayDeaths": 3,
        },
        Object {
          "activeCases": 4324,
          "cases": 12313,
          "country": "Italy",
          "critical": 2345,
          "deaths": 123,
          "recovered": 32523,
          "todayCases": 234,
          "todayDeaths": 23,
        },
      ]
    }
  />
  <hr />
  <NewsBox
    isLoading={false}
    key="Title1"
    newsItem={
      Object {
        "description": "description1",
        "link": "https://link1.com",
        "title": "Title1",
      }
    }
  />
  <NewsBox
    isLoading={false}
    key="Title2"
    newsItem={
      Object {
        "description": "description2",
        "link": "https://link2.com",
        "title": "Title2",
      }
    }
  />
</div>
`;

exports[`Snapshots for CardBody with CircularProgress not isLoading renders CardBody with only CircularProgress when not isLoading desktop 1`] = `
<div>
  <hr />
  <TotalStatBody
    isLoading={false}
    totalStat={
      Object {
        "cases": 254701,
        "deaths": 10447,
        "recovered": 89071,
        "updated": 1584710988792,
      }
    }
  />
  <hr />
  <TableBody
    countriesStat={
      Array [
        Object {
          "activeCases": 6569,
          "cases": 80967,
          "country": "China",
          "critical": 2136,
          "deaths": 3248,
          "recovered": 71150,
          "todayCases": 39,
          "todayDeaths": 3,
        },
        Object {
          "activeCases": 4324,
          "cases": 12313,
          "country": "Italy",
          "critical": 2345,
          "deaths": 123,
          "recovered": 32523,
          "todayCases": 234,
          "todayDeaths": 23,
        },
      ]
    }
  />
  <hr />
  <NewsBox
    isLoading={false}
    key="Title1"
    newsItem={
      Object {
        "description": "description1",
        "link": "https://link1.com",
        "title": "Title1",
      }
    }
  />
  <NewsBox
    isLoading={false}
    key="Title2"
    newsItem={
      Object {
        "description": "description2",
        "link": "https://link2.com",
        "title": "Title2",
      }
    }
  />
</div>
`;

exports[`Snapshots for CardBody with CircularProgress when isLoading renders CardBody with only CircularProgress when isLoading desktop 1`] = `
<div>
  <hr />
  <TotalStatBody
    isLoading={true}
    totalStat={
      Object {
        "cases": 254701,
        "deaths": 10447,
        "recovered": 89071,
        "updated": 1584710988792,
      }
    }
  />
  <hr />
  <WithStyles(ForwardRef(CircularProgress)) />
  <hr />
</div>
`;
